src/main/java/org/fiware/iam/tmforum/PolicyResolver.java

```diff
@@ -0,0 +1,120 @@
+package org.fiware.iam.tmforum;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.http.HttpResponse;
+import jakarta.inject.Singleton;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.fiware.iam.til.model.CredentialsVO;
+import org.fiware.iam.tmforum.productcatalog.api.ProductOfferingApiClient;
+import org.fiware.iam.tmforum.productcatalog.api.ProductSpecificationApiClient;
+import org.fiware.iam.tmforum.productcatalog.model.ProductSpecificationRefVO;
+import org.fiware.iam.tmforum.productcatalog.model.*;
+import org.fiware.iam.tmforum.productorder.model.ProductOfferingRefVO;
+import org.fiware.iam.tmforum.productorder.model.*;
+import org.fiware.iam.tmforum.quote.api.QuoteApiClient;
+import org.fiware.iam.tmforum.quote.model.QuoteItemVO;
+import org.fiware.iam.tmforum.quote.model.QuoteStateTypeVO;
+import org.fiware.iam.tmforum.quote.model.QuoteVO;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * Extract policies from ProductOrders, either from the connected Quote or ProductSpec.
+ */
+@Singleton
+@Slf4j
+@RequiredArgsConstructor
+public class PolicyResolver {
+
+	private static final String AUTHORIZATION_POLICY_KEY = "authorizationPolicy";
+	private static final String QUOTE_DELETE_ACTION = "delete";
+
+	private final ProductOfferingApiClient productOfferingApiClient;
+	private final ProductSpecificationApiClient productSpecificationApiClient;
+	private final QuoteApiClient quoteApiClient;
+	private final ObjectMapper objectMapper;
+
+	public Mono<List<Map<String, Object>>> getAuthorizationPolicy(ProductOrderVO productOrder) {
+		log.info("Get config.");
```

Comment:
We should add some arguments to make the log messages more useful

Reply:
We should remove it, I guess;)